COMPILING.md

```diff
@@ -164,17 +164,41 @@ There is an experimental build based on CMake instead of hand-written
 makefiles. It should work on a wider variety of systems than the standard
 makefile build, and can integrate better with IDEs and static-analysis tools.
 
-1. Run `cmake --version`. If you get a command-not-found error, or the
+1. Ensure you have all the build dependencies installed:
+   - On Debian-like distributions, do
+     ```
+     apt-get install g++ gcc flex bison make git libwww-perl patch
+     ```
+   - On Red Hat/Fedora or derivates, do
+     ```
+     yum install gcc gcc-c++ flex bison perl-libwww-perl patch devtoolset-6
+     ```
+   - On Solaris 11, do
+     ```
+     pkg install system/header developer/lexer/flex developer/parser/bison developer/versioning/git
+     pkg install --accept developer/gcc/gcc-c++-5
+     ```
+   - On FreeBSD 11, do
+     ```
+     pkg install bash gmake git www/p5-libwww patch flex bison
+     ```
+   - On macOS, do
+     ```
+     xcode-select --install
+     ```
+   - On Windows, install Cygwin, then from the Cygwin setup facility install
+    `flex`, `bison`, `tar`, `gzip`, `git`, `make`, `wget`, and `patch`.
+2. Run `cmake --version`. If you get a command-not-found error, or the
    installed version is lower than 3.2, go and install a new version. Most
    Linux distributions have a package for CMake, and Mac users can get it
    through Homebrew.  Windows users should download it manually from cmake.org.
-2. Create a directory to store your build:
```

Comment:
The instructions in the first step should include how to get cmake; then step 2 should simply work.

Reply:
Updated